app/components/threads_button/threads_button.tsx

```diff
@@ -54,32 +59,43 @@ const ThreadsButton = ({currentChannelId, groupUnreadsSeparately, onlyUnreads, u
     const customStyles = getStyleSheet(theme);
 
     const handlePress = useCallback(preventDoubleTap(() => {
-        switchToGlobalThreads(serverUrl);
+        if (onPress) {
+            onPress();
+        } else {
+            switchToGlobalThreads(serverUrl);
+        }
     }), [serverUrl]);
 
     const {unreads, mentions} = unreadsAndMentions;
     const isActive = isTablet && !currentChannelId;
 
-    const [containerStyle, iconStyle, textStyle] = useMemo(() => {
+    const [containerStyle, iconStyle, textStyle, badgeStyle] = useMemo(() => {
         const container = [
             styles.container,
-            isActive ? styles.activeItem : undefined,
+            isActive && styles.activeItem,
         ];
 
         const icon = [
             customStyles.icon,
             isActive || unreads ? customStyles.iconActive : undefined,
+            isInfo ? customStyles.iconInfo : undefined,
```

Comment:
You changed the way it works on the previous style, but not on this one. Any reason not to keep the same pattern?